### PR TITLE
Consolidate pifaces

### PR DESCRIPTION
**Pull request by donaldcampbelljr**
Addresses #493

### CODE REVIEW

**Comment by donaldcampbelljr** [comment]
This PR contains a lot of files because I needed to also pull in the refactored hello_looper-dev changes so that the tests would pass.Therefore, all of the changes under `tests/data` are simple refactoring

**Comment by donaldcampbelljr** [comment]
Also, I believe we wanted to raise an exception if the `pipeline_name` is not the same as the one in the pipestat output schema when using pipestat. Currently, it only warns the user.

File: looper/project.py [hunk 1]
```diff
@@ -512,11 +516,10 @@ def _create_pipestat_config(self, piface):
             pipestat_config_dict.update({"pipeline_name": piface.data["pipeline_name"]})
         else:
             pipeline_name = None
-        if "pipeline_type" in piface.data:
-            pipestat_config_dict.update({"pipeline_type": piface.data["pipeline_type"]})
 
         # Warn user if there is a mismatch in pipeline_names from sources!!!
         if pipeline_name != output_schema_pipeline_name:
+            # TODO Raise exception here.
```
Comment (khoroshevskyi) [inline looper/project.py:522]: leave it as a warning

File: looper/utils.py [hunk 2]
```diff
@@ -606,36 +614,31 @@ def read_looper_config_file(looper_config_path: str) -> dict:
...
-                )
-                if pipeline_type == PipelineLevel.SAMPLE.value:
+            # if pipeline_types is None:
+            #     raise PipelineInterfaceConfigError(
+            #         f"'sample_interface and/or project_interface must be defined in each pipeline interface."
+            #     )
+            # This will append the same, consolidated piface to two different lists
+            # In reality only the command templates are the differentiator
+            if pipeline_types is not None:
+                # TODO should we raise an exception here? I guess you can amend samples with interfaces...
```
Comment (khoroshevskyi) [inline looper/utils.py:633]: I agree, we should add exception here

**Review by khoroshevskyi — APPROVED** [review]
everything looks really good!

File: looper/project.py (hunk 1 above)
Comment (donaldcampbelljr) [reply looper/project.py:522]: OK. I will leave as a warning.

File: looper/utils.py (hunk 2 above)
Comment (donaldcampbelljr) [reply looper/utils.py:785]: Actually, we don't want to do this because it will interfere with this concept: https://github.com/pepkit/looper/issues/26

**Pull request merged**